core/src/main/java/feast/core/config/MonitoringConfig.java

```diff
@@ -23,11 +23,13 @@
 import io.prometheus.client.exporter.MetricsServlet;
 import javax.servlet.http.HttpServlet;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
 import org.springframework.boot.web.servlet.ServletRegistrationBean;
 import org.springframework.context.annotation.Bean;
 import org.springframework.context.annotation.Configuration;
 
 @Configuration
+@ConditionalOnProperty("management.metrics.export.simple.enabled")
```

Comment:
I made it optional to be able to turn off in tests. Since collectors can be cleaned between tests and they conflict

Reply:
Can you try `CollectorRegistry.defaultRegistry.clear();` in your tearDown()?